sysbuild/secureboot.cmake

```diff
@@ -7,34 +7,62 @@ get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
 if(SB_CONFIG_SECURE_BOOT)
   if(SB_CONFIG_SECURE_BOOT_NETCORE)
     set(secure_boot_source_dir ${ZEPHYR_NRF_MODULE_DIR}/samples/nrf5340/netboot)
-  else()
+
+    ExternalZephyrProject_Add(
+      APPLICATION b0n
+      SOURCE_DIR ${secure_boot_source_dir}
+      BOARD ${SB_CONFIG_SECURE_BOOT_NETWORK_BOARD}
+      BUILD_ONLY true
+    )
+    set_target_properties(${SB_CONFIG_SECURE_BOOT_IMAGE_NAME} PROPERTIES
+      IMAGE_CONF_SCRIPT ${CMAKE_CURRENT_LIST_DIR}/image_configurations/b0_image_default.cmake
+    )
+
+    # Can we do this in a more clever way ?
+    # Should adding of hci_rpmasg be a downstream feature, or could we perhaps
+    # overload upstream image adding.
+    if(NOT "CPUNET" IN_LIST PM_DOMAINS)
+      list(APPEND PM_DOMAINS CPUNET)
+    endif()
+    set_property(GLOBAL APPEND PROPERTY
+        PM_CPUNET_IMAGES
+        "b0n"
+    )
+
+    if(SB_CONFIG_SECURE_BOOT_DOMAIN_APP)
+      set_property(GLOBAL PROPERTY DOMAIN_APP_CPUNET
+                   "b0n"
+      )
+    endif()
+endif()
+  if(SB_CONFIG_SECURE_BOOT_APPCORE)
     set(secure_boot_source_dir ${ZEPHYR_NRF_MODULE_DIR}/samples/bootloader)
-  endif()
-  ExternalZephyrProject_Add(
-    APPLICATION ${SB_CONFIG_SECURE_BOOT_IMAGE_NAME}
-    # ToDo: which sample is used as  secure boot sample in other cases ?
-    SOURCE_DIR ${secure_boot_source_dir}
-    BOARD ${SB_CONFIG_SECURE_BOOT_BOARD}
-  )
-  set_target_properties(${SB_CONFIG_SECURE_BOOT_IMAGE_NAME} PROPERTIES
-    IMAGE_CONF_SCRIPT ${CMAKE_CURRENT_LIST_DIR}/image_configurations/b0_image_default.cmake
-  )
-
-  # Can we do this in a more clever way ?
-  # Should adding of hci_rpmasg be a downstream feature, or could we perhaps
-  # overload upstream image adding.
-  if(NOT "${SB_CONFIG_SECURE_BOOT_DOMAIN}" IN_LIST PM_DOMAINS)
-    list(APPEND PM_DOMAINS ${SB_CONFIG_SECURE_BOOT_DOMAIN})
-  endif()
-  set_property(GLOBAL APPEND PROPERTY
-      PM_${SB_CONFIG_SECURE_BOOT_DOMAIN}_IMAGES
-      "${SB_CONFIG_SECURE_BOOT_IMAGE_NAME}"
-  )
-
-  if(SB_CONFIG_SECURE_BOOT_DOMAIN_APP)
-    set_property(GLOBAL PROPERTY DOMAIN_APP_${SB_CONFIG_SECURE_BOOT_DOMAIN}
-                 "${SB_CONFIG_SECURE_BOOT_IMAGE_NAME}"
+
+    ExternalZephyrProject_Add(
+      APPLICATION b0
+      SOURCE_DIR ${secure_boot_source_dir}
+      BUILD_ONLY true
+    )
+    set_target_properties(${SB_CONFIG_SECURE_BOOT_IMAGE_NAME} PROPERTIES
+      IMAGE_CONF_SCRIPT ${CMAKE_CURRENT_LIST_DIR}/image_configurations/b0_image_default.cmake
+    )
+
+    # Can we do this in a more clever way ?
+    # Should adding of hci_rpmasg be a downstream feature, or could we perhaps
+    # overload upstream image adding.
```

Comment:
Should be a JIRA ticket and not a comment here.

Reply:
Removed